website: update the vs. envoy and proxies page (#6326)
hashicorp/consul: website/source/intro/vs/proxies.html.md

Modern service proxies provide high-level service routing, authentication,
telemetry, and more for microservice and cloud environments. Envoy is
a popular and feature rich proxy which Consul [integrates with](https://www.consul.io/docs/connect/proxies/envoy.html) but that is often

> utilized without Consul or something like it.

Reviewer: ```suggestion
used on its own. Consul [integrates with Envoy](https://www.consul.io/docs/connect/proxies/envoy.html) to simplify its configuration. 
```

control plane, Consul integrates with many data plane solutions including
Envoy, HAProxy, Nginx, and more.

The [Consul Envoy integration](https://www.consul.io/docs/connect/proxies/envoy.html)
is currently the primary way to utilize advanced layer 7 features provided
by Consul. In addition to Envoy, Consul enables
third party proxies to integrate with Connect and provide the data
plane with Consul operating as the control plane.

Proxies provide excellent solutions to layer 7 concerns such as path-based
routing, tracing and telemetry, and more. By supporting a pluggable data plane model, the right proxy can be

> deployed as needed. For non-performance critical layer 4 applications, the [built-in proxy](https://www.consul.io/docs/connect/proxies/built-in.html)

Reviewer: ```suggestion
deployed as needed.
```

> could be used. For performance critical applications or those

Reviewer: ```suggestion
For performance-critical applications or those
```

> that utilize layey 7 functionality, Envoy can be used.

Reviewer: ```suggestion
that utilize layey 7 functionality, Envoy can be used. For non-performance critical layer 4 applications, you can use Consul's [built-in proxy](https://www.consul.io/docs/connect/proxies/built-in.html) for convenience.
```

For some applications that may require hardware, a hardware load balancer
such an F5 appliance may be deployed. Consul encourages this use of the right

> proxy for the scenario and treats them as swappable components that can be run

Reviewer: ```suggestion
proxy for the scenario and treats hardware load balancers as swappable components that can be run
```
